lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -173,6 +174,36 @@ impl Drop for InstanceHandle {
 /// [`InstanceHandle`](struct.InstanceHandle.html) smart pointers. This guarantees that instances
 /// and their fields are never moved in memory, otherwise raw pointers in the metadata could be
 /// unsafely invalidated.
+///
+/// An instance occupies one 4096-byte page in memory, with a layout like:
+/// Page {
+///   Instance {
+///     .magic
+///     .embed_ctx
+///      ... etc ...
+///   }
+///
+///   // unused space
+///
+///   InstanceInternals {
+///     .globals
+///     .instruction_counter
+///   }
+/// }
+/// VMContext points right here:
+/// Heap {
+///   ..
+/// }
+///
+/// This layout allows modules to tightly couple to a handful of fields related to the instance,
+/// rather than possibly requiring compiler-side changes (and recompiles) whenever `Instance`
+/// changes.
+///
+/// It also obligates `Instance` to be immediately followed by the heap, but otherwise leaves the
+/// locations of the stack, globals, and any other data, to be implementation-defined by the
+/// `Region` that actually creates `Slot`s onto which `Instance` are mapped.
+/// For information about the layout of all instance-related memory, see the documentation of
+/// `./src/region/mmap.rs#MmapRegion`.
```

Comment:
I need to look up how to do proper doc links, unless someone knows off the top of their head?